skbio/stats/composition.py

```diff
@@ -385,5 +387,257 @@ def centralize(mat):
+       Runs multiple comparisons correction or not.
+       If specified, this will run Holm-Boniferroni
+       correction
+    tau: float
+       A constant used to determine an appropriate
+       cutoff (@wdwvt1 can you comment here plz?)
+    func: function
+       A statistical signficance function to test for
+       signficance between classes.
+       The default is scipy.stats.ttest_ind
+
+    Returns:
+    --------
+    W : pd.Series
+        List of W statistics
+    reject : pd.Series
+        Indicates if the null hypothesis has been rejected
+
+    References
+    ----------
+    ..[1] S. Mandal, 'Analysis of composition of microbiomes:
+          a novel method for studying microbial composition'
+
+    Examples
+    --------
+    First import all of the necessary modules
+
+    >>> from skbio.stats.composition import ancom
+    >>> import pandas as pd
+
+    Now lets load in a pandas dataframe with sample and feature ids
+    for our data matrix.
+    >>> table = pd.DataFrame(
+    ...     [[10., 11., 10., 10., 10., 10., 10.],
+    ...      [10.5, 11.5, 10.5, 10.5, 10.5, 10.5, 10.5],
+    ...      [10., 11., 10., 10., 10., 10., 10.],
+    ...      [20., 21., 10., 10., 10., 10., 10.],
+    ...      [20.5, 21.5, 10.5, 10.5, 10.5, 10.5, 10.5],
+    ...      [20.3, 21.3, 10.2, 10.3, 10.1, 10.6, 10.4]],
+    ...     index=['s1','s2','s3','s4','s5','s6'],
+    ...     columns=['b1','b2','b3','b4','b5','b6','b7'])
+
+    Then create a create a category vector.  In this scenerio, there
+    are only two class, so the first three samples fall under the first
+    class while the last three samples fall under the last class
+    >>> cats = pd.Series([0, 0, 0, 1, 1, 1],
+    ...                  index=['s1','s2','s3','s4','s5','s6'])
+
+    Now run ancom and see if there are any features that have any
+    significant differences
+    >>> W, reject = ancom(table, cats)
+    >>> print(W)
+    b1    7
+    b2    7
+    b3    3
+    b4    3
+    b5    3
+    b6    3
+    b7    3
+    dtype: float64
+    >>> print(reject)
+    b1     True
+    b2     True
+    b3    False
+    b4    False
+    b5    False
+    b6    False
+    b7    False
+    dtype: bool
+    """
+    if len(mat) != len(cats):
+        raise ValueError('The number of samples in mat needs'
+                         'to be the same as the number of samples'
+                         'in cats')
+    mat = _check_composition(mat, ignore_zeros=False)
+    cats = pd.Series(cats)
+
+    mat = mat.sort_index()
+    cats = cats.sort_index()
+    labs = mat.columns
+
+    mat = np.atleast_2d(mat.values)
+    cats = np.array(cats.values)
+
+    _logratio_mat = _log_compare(mat, cats, func)
+    logratio_mat = _logratio_mat + _logratio_mat.T
+
+    n_samp, n_feat = mat.shape
+    # Multiple comparisons
+    if multicorr:
+        for i in range(n_feat):
+            pvalues = _holm(logratio_mat[i, :])
+            logratio_mat[i, :] = pvalues
+
+    W = np.zeros(n_feat)
```

Comment:
I like it.  Consider it done.

On Sun, Oct 11, 2015 at 2:14 PM, Will Van Treuren <notifications@github.com>
wrote:

> In skbio/stats/composition.py
> <https://github.com/biocore/scikit-bio/pull/1114#discussion_r41715465>:
>
> > +    labs = mat.columns
> > +
> > +    mat = np.atleast_2d(mat.values)
> > +    cats = np.array(cats.values)
> > +
> > +    _logratio_mat = _log_compare(mat, cats, func)
> > +    logratio_mat = _logratio_mat + _logratio_mat.T
> > +
> > +    n_samp, n_feat = mat.shape
> > +    # Multiple comparisons
> > +    if multicorr:
> > +        for i in range(n_feat):
> > +            pvalues = _holm(logratio_mat[i, :])
> > +            logratio_mat[i, :] = pvalues
> > +
> > +    W = np.zeros(n_feat)
>
> could this be shortened to something like:
>
> W = (logratio_mat < alpha).sum(1)
>
> This is stylistic - feel free to ignore if you don't like it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/scikit-bio/pull/1114/files#r41715465>.
>